src/blssignature.cpp

```diff
@@ -36,35 +36,134 @@ BLSSignature BLSSignature::FromBytes(const uint8_t *data) {
     return sigObj;
 }
 
-BLSSignature BLSSignature::FromBytes(const uint8_t *data,
-                                     const AggregationInfo &info) {
+BLSInsecureSignature BLSInsecureSignature::FromG2(const relic::g2_t* element) {
+    BLS::AssertInitialized();
+    BLSInsecureSignature sigObj = BLSInsecureSignature();
+    relic::g2_copy(sigObj.sig, *(relic::g2_t*)element);
+    return sigObj;
+}
+
+BLSInsecureSignature::BLSInsecureSignature() {
+    BLS::AssertInitialized();
+    g2_set_infty(sig);
+}
+
+BLSInsecureSignature::BLSInsecureSignature(const BLSInsecureSignature &signature) {
+    BLS::AssertInitialized();
+    g2_copy(sig, *(relic::g2_t*)&signature.sig);
+}
+
+void BLSInsecureSignature::GetPoint(relic::ep2_st* output) const {
+    *output = *sig;
+}
+
+BLSInsecureSignature BLSInsecureSignature::Aggregate(const BLSInsecureSignature& r) const {
+    BLSInsecureSignature result(*this);
+    g2_add(result.sig, *(relic::g2_t*)&result.sig, *(relic::g2_t*)&r.sig);
+    return result;
+}
+
+BLSInsecureSignature BLSInsecureSignature::Aggregate(const std::vector<BLSInsecureSignature>& sigs) {
```

Comment:
Can we rename these two methods to Mul? All it's doing is multiplying the signatures together.

Reply:
It is also the opposite of divide, so it make sense